core/src/main/scala/cats/data/NonEmptyList.scala

```diff
@@ -376,40 +376,79 @@ final case class NonEmptyList[+A](head: A, tail: List[A]) extends NonEmptyCollec
    * res0: Boolean = true
    * }}}
    */
-  def groupBy[B](f: A => B)(implicit B: Order[B]): SortedMap[B, NonEmptyList[A]] = {
-    implicit val ordering: Ordering[B] = B.toOrdering
-    var m = TreeMap.empty[B, mutable.Builder[A, List[A]]]
+  def groupBy[K](key: A => K)(implicit K: Order[K]): SortedMap[K, NonEmptyList[A]] =
```

Comment:
I am not sure if the change of the name of the generic and the function can affect binary compatibility.
And if the break of source compatibility is acceptable.